### PR TITLE
add validator

### PR DESCRIPTION
**Pull request by kaallaash**

### CODE REVIEW

File: DescriptiveGeometry/DG.API/Controllers/DrawingController.cs [hunk 1]
```diff
@@ -62,6 +76,16 @@ public class DrawingController : Controller
         [FromBody] ChangeDrawingViewModel changeDrawingViewModel,
       CancellationToken cancellationToken)
     {
+        var validationResult =
+            await _changeDrawingViewModelValidator
+                .ValidateAsync(changeDrawingViewModel, cancellationToken);
```
Comment (moltenessence) [inline DescriptiveGeometry/DG.API/Controllers/DrawingController.cs:81]: You can use ValidateAndThrowAsync to get rid of your extensions

**Review by moltenessence — CHANGES_REQUESTED** [review]

File: DescriptiveGeometry/DG.API/Validators/ChangeDrawingViewModelValidator.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿using DG.API.ViewModels;
+using FluentValidation;
+
+namespace DG.API.Validators;
+
+public class ChangeDrawingViewModelValidator : AbstractValidator<ChangeDrawingViewModel>
+{
+    public ChangeDrawingViewModelValidator()
+    {
+        RuleFor(d => d.Description).NotNull();
```
Comment (Blackolf931) [inline DescriptiveGeometry/DG.API/Validators/ChangeDrawingViewModelValidator.cs:10]: Is a string consisting of spaces processed here?

**Review by Blackolf931 — CHANGES_REQUESTED** [review]

File: DescriptiveGeometry/DG.API/Controllers/DrawingController.cs [hunk 3]
```diff
@@ -48,6 +51,9 @@ public class DrawingController : Controller
         [FromBody] ChangeDrawingViewModel changeDrawingViewModel,
         CancellationToken cancellationToken)
     {
+        await _changeDrawingViewModelValidator
+            .ValidateAndThrowAsync(changeDrawingViewModel, cancellationToken)
```
[the remark this answers is not in the document]
Comment (kaallaash) [reply DescriptiveGeometry/DG.API/Controllers/DrawingController.cs]: my fail

**Review by moltenessence — CHANGES_REQUESTED** [review]
Check out your ci/cd, one test doesn't pass

**Comment by kaallaash** [comment]
> Check out your ci/cd, one test doesn't pass

I did hotfix PR for it. https://github.com/nantonov/AKD/pull/26

**Review by nantonov — APPROVED** [review]

**Review by stas1408123 — APPROVED** [review]

**Review by moltenessence — APPROVED** [review]

**Pull request merged**